### PR TITLE
check dom node exists

### PR DESCRIPTION
**Pull request by mloginov**
fix to prevent editor crash if monaco dom node is removed before constrained editor instance is removed

### CODE REVIEW

File: src/constrainedEditor.js [hunk 1]
```diff
@@ -48,9 +48,9 @@ export function constrainedEditor(monaco) {
       domNode.addEventListener('keydown', manipulator._listener, true);
       editorInstance.onDidChangeModel(function () {
         // domNode - refers old dom node
-        domNode.removeEventListener('keydown', manipulator._listener, true)
+        domNode && domNode.removeEventListener('keydown', manipulator._listener, true)
         const newDomNode = editorInstance.getDomNode(); // Gets Current dom node
-        newDomNode.addEventListener('keydown', manipulator._listener, true);
+        newDomNode && newDomNode.addEventListener('keydown', manipulator._listener, true);
```
Comment (Pranomvignesh) [inline src/constrainedEditor.js:53]: Hi @mloginov, These listeners are inside `onDidChangeModel` event callback, if the dom node is **accidentally** removed, then the I guess there is no possibility of monaco **automatically** triggering onDidChangeModel function (unless we tweak it to trigger, which is not an ideal case)

Yes, There will be a memory leak issue as manipulator._listener function is not removed from the node. But I'm expecting the user to call the disposeConstrainer method before node deletion.

Also, I cannot handle accidental node deletion as it is costly to watch for DOM node removal, else this event listener removal could be done in that callback.

**I request you to provide me a regular use case where this check helps to prevent crashing of the program**

Comment (mloginov) [reply src/constrainedEditor.js:53]: Example 
You have 3 containers, 3 different models and one monaco editor instance
Container 1 and 3 have constrained editor instance.

Tab definition something like that

```
class Tab {
  constructor(editorInstance, containerId) {
    this.editorInstance = editorInstance;
    this.containerId = containerId;
    this.model = window.monaco.editor.createModel("some content");
  }

  init() {
    // attach editor
    this.constrainedInstance = constrainedEditor(window.monaco);
    this.editorInstance.setModel(this.model)
    this.constrainedInstance.initializeIn(this.editorInstance);
    this.constrainedInstance.addRestrictionsTo(this.model, [
      {
        range: [1, 2, 1, 4] // spme range
      }
    ]);
  }

  destroy() {
    this.constrainedInstance.disposeConstrainer();
    this.constrainedInstance = null
    // detach editor
  }
}
```

When you switch tabs, you must detach the editor from the old container and then attach to the new container and initialize it.

```
function switchTab() {
  currentTab.detach()
  newTab.init()
}
```

In this case, we may have some side effects because `onDidChangeModel` is not removed when disposeConstrainer is called.

Use case - IDE. Panel can have a different tabs, and it has 1 editor for panel. Switch between tabs. Also when tab with the editor closes we need to detach model(for example switch editor model to dummy)
```setModel(window.monaco.editor.createModel("");)``` - it also triggers `onDidChangeModel` 

So i think it would be great if null checks will be added(`getDomNode` can return null) and `onDidChangeModel` will be removed on `disposeConstrainer` called

Comment (mloginov) [reply src/constrainedEditor.js:53]: Removing `onDidChangeModel` when `disposeConstrainer` is called will be enough. And perhaps add to the documentation that `disposeConstrainer` should be called before the editor is removed from the DOM.

Comment (Pranomvignesh) [reply src/constrainedEditor.js:53]: >Use case - IDE. Panel can have a different tabs, and it has 1 editor for panel. Switch between tabs. Also when tab with the editor closes we need to detach model(for example switch editor model to dummy)
setModel(window.monaco.editor.createModel("");) - it also triggers onDidChangeModel

@mloginov  Thanks for pointing this out.
Still this `setModel` will be triggered by users and not by monaco automatically.
Anyway adding a check will make it more stable.
Thanks for your suggestion.
Will merge it

**Review by Pranomvignesh — APPROVED** [review]

**Comment by Pranomvignesh** [comment]
@mloginov You commits aren't verified 
Please refer this [link](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification)

I'm not able to merge it
Screen Shot 2022-09-12 at 1 55 56 PM

**Comment by mloginov** [comment]
> @mloginov You commits aren't verified Please refer this [link](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification)
> 
> I'm not able to merge it Screen Shot 2022-09-12 at 1 55 56 PM

Fixed

**Comment by Pranomvignesh** [comment]
Thanks @mloginov
